packages/scm/src/browser/scm-amend-widget.tsx

```diff
@@ -51,11 +50,11 @@ export class ScmAmendWidget extends ReactWidget {
         this.id = ScmAmendWidget.ID;
     }
 
-    protected onUpdateRequest(msg: Message): void {
-        if (!this.isAttached || !this.isVisible) {
-            return;
-        }
-        super.onUpdateRequest(msg);
+    @postConstruct()
+    protected init(): void {
```

Comment:
Is it necessary? Should not `update` be called from `ScmWidget.refresh` anyway? And `ScmWidget.refresh` already called on `onDidChangeSelectedRepository`?

Reply:
> Is it necessary? Should not `update` be called from `ScmWidget.refresh` anyway? And `ScmWidget.refresh` already called on `onDidChangeSelectedRepository`?

You are correct that it is technically not necessary.  However surely if a widget is getting a repository directly from ScmService then the widget should also be listening for changes.  Widgets should work on their own and not be dependent on someone outside knowing that the widget needs updating on repository change.

On the other hand the ScmCommit and ScmAmend widgets are not really re-usable so I'm happy to change.  I have pushed a commit to remove from both.